metals/src/main/scala/scala/meta/internal/metals/BuildServerConnection.scala

```diff
@@ -279,16 +289,35 @@ object BuildServerConnection {
       }
     }
 
-    setupServer().map { connection =>
-      new BuildServerConnection(
-        setupServer,
-        connection,
-        languageClient,
-        reconnectNotification,
-        config,
-        workspace
-      )
-    }
+    setupServer()
+      .map { connection =>
+        new BuildServerConnection(
+          setupServer,
+          connection,
+          languageClient,
+          reconnectNotification,
+          config,
+          workspace
+        )
+      }
+      .recoverWith { case NonFatal(e) =>
```

Comment:
Minor, but while in `setupServer` we're explicitly hanlding `TimeoutException` (for then rethrowing it) here we're catching all non fatals. Should we limit ourselves to `TimeoutException` here too? Do you expect other errors happening here?

Reply:
Right, `TimeoutException` should be actually the only thing that gets thrown here. Fixed!